tests/drivers/gpio/gpio_api_1pin/boards/bl654_dvk.overlay

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2021 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	resources {
+		compatible = "test,gpio_basic_api";
```

Comment:
Let's test the driver with bt6x0 or bt6x0 board, not bl654_dvk. It's OK that the boards are added in a later commit.

The file is based on `tests/drivers/gpio/gpio_api_1pin/boards/particle_xenon.overlay` but this is actually not correct. The gpio_api_1pin testcase relies on the definition of led0 alias in a board file. Since virtually all boards define it there is typically no need for extra .overlay and .conf files in the board directory. The testcase should just work. Both `tests/drivers/gpio/gpio_api_1pin/boards/bl654_dvk.conf` and `tests/drivers/gpio/gpio_api_1pin/boards/bl654_dvk.overlay` can be removed.

Reply:
Updated as discussed with both test cases passing. I needed to add some additional functionality, not realising that interrupts could be triggered when outputs change state.

Reply:
I've removed the particle_xenon files and replaced with bl5340 equivalents. Both tests are passing, with one board needing to be modifed to short an expander output to expander input.